### DISCUSSION TITLE
Call to engineer a platform where we can gain access to Windows, Macintosh and Debian operating systems

### DISCUSSION THREAD

**Discussion by Elvis254** [dotnet/roslyn#74900] [Ideas] [answered]
Hi,
I would like a platform where us software engineers can gain access to Windows, Macintosh and Debian operating systems so as build software installation files without requiring someone to acquire a physical device to be engineered and made available on the cloud. Here is a genuine reason why we really need the said platform: [https://www.electronforge.io/](https://www.electronforge.io/)
Elvis.

**Comment by 333fred** [comment]
This doesn't seem related to the Roslyn compiler, closing out.

**Reply by Elvis254** [reply]
@333fred please, take a look at it because this new technology requires it. Remember we use C# which utilizes the Roslyn compiler. [https://learn.microsoft.com/en-us/dotnet/maui/deployment/?view=net-maui-8.0](https://learn.microsoft.com/en-us/dotnet/maui/deployment/?view=net-maui-8.0)

**Reply by 333fred** [reply, marked as answer]
@Elvis254 it's unclear what you're asking for. Roslyn produces standard .NET binaries: these are already operating-system independent, and mostly CPU-architecture independent; where they aren't aren't, there are already systems in MSBuild for controlling this. There doesn't appear to be an ask for Roslyn to do anything that it doesn't already do.

**Reply by Elvis254** [reply]
@333fred well according to this document [https://learn.microsoft.com/en-us/dotnet/maui/ios/pair-to-mac?view=net-maui-8.0](https://learn.microsoft.com/en-us/dotnet/maui/ios/pair-to-mac?view=net-maui-8.0) its possible to build an iOS app if the owner accepts your request to use it. A big question comes in what if the owner refuses to give you access to their computer? Please, share with me a link where I can post this pleasant idea of mine.

**Reply by 333fred** [reply]
This isn't something that Microsoft can help with. Apple building requirements are enforced by them. If you don't have an Apple device, you can rent one through various services like https://www.macincloud.com.
